src/compiler/typescript_to_rust/azle_generate/src/generators/boa_error_handlers.rs

```diff
@@ -3,21 +3,21 @@ use quote::quote;
 
 pub fn generate() -> TokenStream {
     quote! {
-        pub fn _azle_handle_boa_result(
+        pub fn _azle_unwrap_boa_result(
             boa_result: boa_engine::JsResult<boa_engine::JsValue>,
             context: &mut boa_engine::Context
         ) -> boa_engine::JsValue {
             match boa_result {
                 Ok(_azle_boa_return_value) => _azle_boa_return_value,
                 Err(_azle_boa_error) => {
-                    let error_message = _azle_handle_boa_error(_azle_boa_error.to_opaque(context), context);
+                    let error_message = _azle_js_value_to_string(_azle_boa_error.to_opaque(context), context);
 
-                    panic!("Azle runtime error: {}", error_message);
+                    ic_cdk::api::trap(&format!("Uncaught {}", error_message));
                 },
             }
         }
 
-        fn _azle_handle_boa_error(error_value: boa_engine::JsValue, context: &mut boa_engine::Context) -> String {
+        fn _azle_js_value_to_string(error_value: boa_engine::JsValue, context: &mut boa_engine::Context) -> String {
```

Comment:
I don't think this is an appropriate name, it sounds like it's a generic JsValue -> String function. It's specifically turning a JS error into a string